### PR TITLE
Add makedirs to file operations

### PR DESCRIPTION
**Pull request by montanaflynn**
### Background

Many times auto gpt will try to download or append to files in a directory that doesn't exist, this will cause an error that it then tries to correct, sometimes unsuccessfully. 

Related issue: https://github.com/Significant-Gravitas/Auto-GPT/issues/3249

### Changes

I added logic to check if the directory exists and if not create it. I used the same code already being used for `write_to_file` and added it to the `append_to_file` and `download_file` operations.

### Documentation

No documentation.

### Test Plan

I had auto gpt download and append to files.

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes 

I didn't add any tests because I couldn't find out where or how to add them as I found other file operations didn't have tests.

### CODE REVIEW

**Review by hdkiller — DISMISSED** [review]

**Comment by montanaflynn** [comment]
Added tests

File: autogpt/commands/file_operations.py [hunk 1]
```diff
@@ -167,6 +167,9 @@ def append_to_file(filename: str, text: str, should_log: bool = True) -> str:
         str: A message indicating success or failure
     """
     try:
+        directory = os.path.dirname(filename)
+        if not os.path.exists(directory):
+            os.makedirs(directory)
```
Comment (Pwuts) [inline autogpt/commands/file_operations.py]:
```suggestion
        os.makedirs(directory, exist_ok=True)
```

File: autogpt/commands/file_operations.py [hunk 2]
```diff
@@ -236,6 +239,9 @@ def download_file(url, filename):
         filename (str): Filename to save the file as
     """
     try:
+        directory = os.path.dirname(filename)
+        if not os.path.exists(directory):
+            os.makedirs(directory)
```
Comment (Pwuts) [inline autogpt/commands/file_operations.py]:
```suggestion
        os.makedirs(directory, exist_ok=True)
```

File: tests/unit/test_file_operations.py [hunk 3]
```diff
@@ -33,6 +33,7 @@ def setUp(self):
         self.test_file = str(self.workspace.get_path("test_file.txt"))
         self.test_file2 = "test_file2.txt"
         self.test_directory = str(self.workspace.get_path("test_directory"))
+        self.test_nested_file = os.path.join(self.test_directory, self.test_file)
```
Comment (Pwuts) [inline tests/unit/test_file_operations.py]: Why this change? It does an unsafe path join that doesn't have any effect because `self.test_file` is absolute.

**Review by Pwuts — DISMISSED** [review]

File: tests/unit/test_file_operations.py (hunk 3 above)
Comment (montanaflynn) [reply tests/unit/test_file_operations.py]: I was trying to make a nested file so it would test creating the directory, I changed it to:

```python
self.test_nested_file = str(self.workspace.get_path("nested/test_file.txt"))
```

**Review by collijk — APPROVED** [review]
